modules/apps/frontend-js/frontend-js-spa-web/test/app/App.ts

```diff
@@ -1922,49 +2140,58 @@ describe('App', function () {
 				'screenId1',
 				undefined
 			);
-			expect(surface2.getChild('default').innerHTML).toBe('default2');
 
-			this.app.navigate('/path2').then(() => {
+			expect(surface2.getChild('default')?.innerHTML).toBe('default2');
+
+			app?.navigate('/path2').then(() => {
 				expect(surface1.addContent).toHaveBeenCalledWith(
 					'screenId2',
 					undefined
 				);
-				expect(surface1.getChild('default').innerHTML).toBe('default1');
+
+				expect(surface1.getChild('default')?.innerHTML).toBe(
+					'default1'
+				);
 
 				expect(surface2.addContent).toHaveBeenCalledWith(
 					'screenId2',
 					'content2'
 				);
 
-				surface1.getElement().remove();
-				surface2.getElement().remove();
+				surface1.getElement()?.remove();
+				surface2.getElement()?.remove();
+
 				done();
 			});
 		});
 	});
 
 	it('adds surface content after history path is updated', (done) => {
-		var surface = new Surface('surfaceId');
+		const surface = new Surface('surfaceId');
+
 		surface.addContent = () => {
 			expect(window.location.pathname).toBe('/path');
 		};
-		this.app = new App();
-		this.app.addRoutes(new Route('/path', Screen));
-		this.app.addSurfaces(surface);
-		this.app.navigate('/path').then(() => {
+
+		app = new App();
+
+		app.addRoutes(new Route('/path', Screen));
+		app.addSurfaces(surface);
+		app.navigate('/path').then(() => {
 			done();
 		});
 	});
 
 	it.skip('navigates cancelling navigation to multiple paths after navigation is scheduled to keep only the last one', (done) => {
-		const app = (this.app = new App());
+		const app = new App();
 
 		class TestScreen extends Screen {
-			evaluateStyles(surfaces) {
+			evaluateStyles() {
 				userEvent.click(enterDocumentLinkElement('/path2'));
+
 				exitDocumentLinkElement();
 
-				return super.evaluateStyles(surfaces);
+				return super.evaluateStyles();
 			}
 
 			evaluateScripts(surfaces) {
```

Comment:
`surfaces` has typing issues, setting it to `surfaces: Surfaces` throws an error on the `evaluateScripts` function:
```
Property 'evaluateScripts' in type 'TestScreen' is not assignable to the same property in base type 'Screen'.
  Type '(surfaces: Surface) => Promise<void>' is not assignable to type '(surfaces: Surfaces) => Promise<void>'.
    Types of parameters 'surfaces' and 'surfaces' are incompatible.
      Type 'Surfaces' is missing the following properties from type 'Surface': activeChild, defaultChild, element, id, and 18 more.ts(2416)
```

Also `Property 'scheduledNavigationEvent' does not exist on type 'App'. Did you mean 'scheduledNavigationQueue'?`... Not sure if that's a preexisting bug or what, very confusing.